Source/WebCore/page/ScreenOrientation.cpp

```diff
@@ -154,6 +160,9 @@ ExceptionOr<void> ScreenOrientation::unlock()
     if (!document || !document->isFullyActive())
         return Exception { InvalidStateError, "Document is not fully active."_s };
 
+    if (document->page() && !document->page()->isVisible())
+        return Exception { SecurityError, "Only visible documents can unlock the screen orientation"_s };
```

Comment:
> If the document did lock the screen, shouldn't it be able to unlock it whenever it wants? Or maybe, it is somehow not possible, in which case do we need that check?

I was thinking of a case where the document goes fullscreen, switches orientation, then the user switches to another application (but the web app keeps running in the background).

It could unlock, but then we would need to hold that state and do the right thing when it's visible.

> If we want that check, should it be SecurityError or InvalidStateError?

It felt like a `SecurityError`, as I was treating it as a "pre-lock condition", which are `SecurityError`.  

> Also we are doing this check before the isSameOriginAsTopDocument check in unlock but after in lock. Should we be consistent?

Oops, yes. Will fix. 

As an aside, the `isSameOriginAsTopDocument` will eventually go away. That shouldn't be there. But it does need a different check, so should be consistent.